speechbrain/core.py

```diff
@@ -296,16 +315,11 @@ def parse_arguments(arg_list=None):
         help="Use dynamic shape tracing for compilation",
     )
     parser.add_argument(
-        "--auto_mix_prec",
-        default=None,
-        action="store_true",
-        help="This flag enables training with automatic mixed-precision.",
-    )
-    parser.add_argument(
-        "--bfloat16_mix_prec",
-        default=None,
-        action="store_true",
-        help="This flag enables training with bfloat16 mixed-precision.",
+        "--autocast",
```

Comment:
As for naming, I would prefer the old `auto_mix_prec` flag and `AutoMixPrecConfig` or `AMPConfig` for a few reasons:
- the meaning of `autocast` is less intuitive than `auto_mix_prec` for non-expert users
- if we keep the old `auto_mix_prec` flag name, it might be easier to make the new changes backward compatible at some extent
- automatic mixed precision and autocast are not exactly the same thing. You need both autocast and a gradient scaler to implement AMP training. Furthermore, you could train in standard precision and use autocast at inference time only to reduce memory/latency (https://pytorch.org/tutorials/recipes/recipes/amp_recipe.html#inference-evaluation). In our case the goal of this flag is to enable AMP training, but not to enforce a specific precision for validation or test.